### PR TITLE
review/jm-intial;

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This is not an actual PR and is only intended to be used as an initial review of the source build. This PR should be closed following all comments being addressed and replicated in the `main` branch.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: 🐞 Bug
 description: Report a bug or an issue you've found within the dbt package
 title: "[Bug] <title>"
-labels: ["bug", "triage"]
+labels: ["bug", "triage"]
```
Comment (fivetran-joemarkiewicz) [inline .github/ISSUE_TEMPLATE/bug-report.yml:4]: Can we update these labels to be the new ones outlined for our github reporting and tracking.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 contact_links:
-  - name: Ask a question during our office hours
-    url: https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours
-    about: Schedule time during the external office hours block with the Fivetran Analytics Engineering team for support
+  - name: Ask a question during our office hours 
+    url: https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours 
+    about: Schedule time during the external office hours block with the Fivetran Analytics Engineering team for support
```
Comment (fivetran-joemarkiewicz) [inline .github/ISSUE_TEMPLATE/config.yml:4]: We are going to start removing the references to office hours in place of directing folks to the dbt user group community. Would you be able to make that update here and within the references section in the README

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: 🎉 Feature
 description: Suggest a new feature for the Fivetran dbt package
 title: "[Feature] <title>"
-labels: ["enhancement"]
+labels: ["enhancement"]
```
Comment (fivetran-joemarkiewicz) [inline .github/ISSUE_TEMPLATE/feature-request.yml:4]: Can we update these labels to be the new ones outlined for our github reporting and tracking.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
-# dbt_Recharge_source v0.1.0
+# dbt_recharge_source v0.1.0
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: Small casing request to edit

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 🎉 This is the initial release of this package! 🎉
 ## 📣 What does this dbt package do?
-- Materializes [Recharge staging tables](https://fivetran.github.io/dbt_Recharge_source/#!/overview/Recharge_source/models/?g_v=1&g_e=seeds), which leverage data in the format described by [this ERD](https://fivetran.com/docs/applications/Recharge#schemainformation). These staging tables clean, test, and prepare your Recharge data from [Fivetran's connector](https://fivetran.com/docs/applications/Recharge) for analysis by doing the following:
+- Materializes [Recharge staging tables](https://fivetran.github.io/dbt_Recharge_source/#!/overview/Recharge_source/models/?g_v=1&g_e=seeds), which leverage data in the format described by [this ERD](https://fivetran.com/docs/applications/Recharge#schemainformation). These staging tables clean, test, and prepare your Recharge data from [Fivetran's connector](https://fivetran.com/docs/applications/Recharge) for analysis by doing the following:
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:4]: The Fivetran links in this block are producing 404 errors. I think it is due to the capital R's. Can we adjust these to ensure they point to the correct docs.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
   - Names columns for consistency across all packages and for easier analysis
   - Adds freshness tests to source data
   - Adds column-level testing where applicable. For example, all primary keys are tested for uniqueness and non-null values.
-- Generates a comprehensive data dictionary of your Recharge data through the [dbt docs site](https://fivetran.github.io/dbt_Recharge_source/).
-- These tables are designed to work simultaneously with our [Recharge transformation package](https://github.com/fivetran/dbt_Recharge).
+- Generates a comprehensive data dictionary of your Recharge data through the [dbt docs site](https://fivetran.github.io/dbt_recharge_source/).
+- These tables are designed to work simultaneously with our [Recharge transformation package](https://github.com/fivetran/dbt_recharge).
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:9]: Small casing change request

File: README.md [hunk 7]
```diff
@@ -17,7 +17,7 @@
   - Name columns for consistency across all packages and easier analysis
   - Adds freshness tests to source data
   - Adds column-level testing where applicable. For example,  all primary keys are tested for uniqueness and non-null values.
-- Generates a comprehensive data dictionary of your Recharge data through the [dbt docs site](https://fivetran.github.io/dbt_recharge_source/).
+- Generates a comprehensive data dictionary of your Recharge data through the [dbt docs site](https://fivetran.github.io/dbt_recharge_source/).
```
Comment (fivetran-joemarkiewicz) [inline README.md:20]: It looks like the docs were generated with the dbt version that doesn't rended the overview page. This just shows a loading page. Would you be able to re-generate with the latest version of dbt to ensure the loading error doesn't persist.

File: README.md [hunk 8]
```diff
@@ -62,10 +62,10 @@ If you would like to pass through additional columns to the staging models, add
 ```yml
 vars:
     recharge__address_passthrough_columns: 
-      - name: "new_custom_field"
-        alias: "custom_field_name"
+      - name: "new_custom_field" 
+        alias: "custom_field_name" 
       - name: "a_second_field"
-    # a similar pattern can be applied to the rest of the following variables.
+    # a similar pattern can be applied to the rest of the following variables.
```
Comment (fivetran-joemarkiewicz) [inline README.md:68]: I really like this approach to not duplicate the docs unnecessarily!

File: README.md [hunk 9]
```diff
@@ -62,10 +62,10 @@ If you would like to pass through additional columns to the staging models, add
 ```yml
 vars:
     recharge__address_passthrough_columns: 
-      - name: "new_custom_field"
-        alias: "custom_field_name"
+      - name: "new_custom_field" 
+        alias: "custom_field_name"
```
Comment (fivetran-joemarkiewicz) [inline README.md:66]: There is also a `transform_sql` argument, do we want to highlight that use case here as well?

File: dbt_project.yml [hunk 10]
```diff
@@ -23,14 +23,14 @@ vars:
...
-    recharge__subscription_passthrough_columns: []
-    recharge__subscription_history_passthrough_columns: []
+    recharge__address_passthrough_columns: [] 
+    recharge__charge_passthrough_columns: [] 
+    recharge__charge_line_item_passthrough_columns: [] 
+    recharge__order_passthrough_columns: [] 
+    recharge__order_line_passthrough_columns: [] 
+    recharge__product_passthrough_columns: [] 
+    recharge__subscription_passthrough_columns: [] 
+    recharge__subscription_history_passthrough_columns: []
```
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:33]: We want to remove these empty variables from the package dbt_project.yml. We found that there are some edge cases where dbt will use these instead of the customer provided values. 

The only thing we need to ensure is that default values are provided when we call the variable.

File: integration_tests/.gitignore [hunk 11]
```diff
@@ -3,4 +3,5 @@ target/
 dbt_modules/
 logs/
 .DS_Store
-dbt_packages/
\ No newline at end of file
+dbt_packages/
+env/
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/.gitignore:7]: In case we are using a virtual env, we should add this to the gitignore

File: integration_tests/dbt_project.yml [hunk 12]
```diff
@@ -26,9 +26,9 @@ vars:
     recharge_subscription_identifier: "subscription_data"
     recharge_subscription_history_identifier: "subscription_history_data"
 
-dispatch:
-  - macro_namespace: dbt_utils
-    search_order: ['spark_utils', 'dbt_utils']
+dispatch: 
+  - macro_namespace: dbt_utils 
+    search_order: ['spark_utils', 'dbt_utils']
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/dbt_project.yml:31]: Is there a reason we do this dispatch twice?

File: macros/.gitkeep [hunk 13]
```diff
@@ -0,0 +1 @@
+
```
Comment (fivetran-joemarkiewicz) [inline macros/.gitkeep:1]: Is this file needed?

File: models/src_recharge.yml [hunk 14]
```diff
@@ -9,9 +9,9 @@ sources:
     loader: Fivetran
     loaded_at_field: _fivetran_synced
 
-    freshness: 
-      warn_after: {count: 48, period: hour}
-      error_after: {count: 168, period: hour}
+    freshness:  
+      warn_after: {count: 48, period: hour} 
+      error_after: {count: 168, period: hour}
```
Comment (fivetran-joemarkiewicz) [inline models/src_recharge.yml:14]: This may be an appropriate freshness, but do you feel this is applicable for **all** sources in this package? For example, I don't forsee the address records changing consistently enough to fit within that time period.

File: models/stg_recharge__address.sql [hunk 15]
```diff
@@ -33,10 +33,12 @@ final as (
         country,
         company,
         phone,
-        cart_attribute_elevar_visitor_info as customer_visitor_info,
+        cart_attribute_elevar_visitor_info as customer_visitor_info,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__address.sql:36]: This looks like it is a custom field. I would prefer to leave out custom fields from the staging model as they may not be applicable to all our users. Instead we can have the passthrough variables perform this feature for the customers that want the extra fields.

File: models/stg_recharge__address.sql [hunk 16]
```diff
@@ -33,10 +33,12 @@ final as (
         country,
         company,
         phone,
-        cart_attribute_elevar_visitor_info as customer_visitor_info,
+        cart_attribute_elevar_visitor_info as customer_visitor_info, 
         _fivetran_deleted,
         cast(_fivetran_synced as {{ dbt.type_timestamp() }}) as _fivetran_synced
+
         {{ fivetran_utils.fill_pass_through_columns('recharge__address_passthrough_columns') }}
+
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__address.sql:41]: Small style update request. Can we have the macro exist on its own. This just helps for readability and for us to understand the macro is being called more easily.

File: models/stg_recharge__address_shipping_line.sql [hunk 17]
```diff
@@ -25,8 +25,8 @@ final as (
         price,
         code,
         title,
-        _fivetran_deleted,
-        cast(_fivetran_synced as {{ dbt.type_timestamp() }}) as _fivetran_synced
+        _fivetran_deleted,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__address_shipping_line.sql:28]: Are we going to want to filter out the deleted records?

File: models/stg_recharge__address_shipping_line.sql [hunk 18]
```diff
@@ -25,8 +25,8 @@ final as (
         price,
         code,
         title,
-        _fivetran_deleted,
-        cast(_fivetran_synced as {{ dbt.type_timestamp() }}) as _fivetran_synced
+        _fivetran_deleted, 
+        cast(_fivetran_synced as {{ dbt.type_timestamp() }}) as _fivetran_synced
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__address_shipping_line.sql:29]: Will we use this field at all downstream? Typically we don't use the fivetran_synced field and we can ommit it from the staging models.

If we are going to remove, let's remove it from all the staging models.

File: models/stg_recharge__charge.sql [hunk 19]
```diff
@@ -50,24 +50,24 @@ final as (
...
+        billing_address_phone, 
+        shipping_address_first_name || ' ' || shipping_address_last_name as shipping_address_full_name, 
+        shipping_address_address_1 as shipping_address_line_1, 
+        shipping_address_address_2 as shipping_address_line_2, 
+        shipping_address_city, 
+        shipping_address_province, 
+        shipping_address_zip, 
+        shipping_address_country, 
+        shipping_address_company, 
+        shipping_address_phone,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__charge.sql:70]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__charge_line_item.sql [hunk 20]
```diff
@@ -25,18 +25,18 @@ final as (
         subscription_id,
         vendor,
         title,
-        property_title as title_v2,
+        property_title as title_v2,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__charge_line_item.sql:28]: Why is this renamed to title_v2?

File: models/stg_recharge__charge_line_item.sql [hunk 21]
```diff
@@ -25,18 +25,18 @@ final as (
...
         shopify_product_id,
         shopify_variant_id,
-        property_charge_interval_frequency as charge_interval_frequency,
-        property_charge_interval_unit_type as charge_interval_unit,
-        property_shipping_interval_frequency as shipping_interval_frequency,
-        property_shipping_interval_unit_type as shipping_interval_unit,
+        property_charge_interval_frequency as charge_interval_frequency, 
+        property_charge_interval_unit_type as charge_interval_unit, 
+        property_shipping_interval_frequency as shipping_interval_frequency, 
+        property_shipping_interval_unit_type as shipping_interval_unit,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__charge_line_item.sql:39]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__discount.sql [hunk 22]
```diff
@@ -38,8 +38,8 @@ final as (
         duration,
         duration_usage_limit,
         minimum_order_amount,
-        once_per_customer,
-        applies_once,
+        once_per_customer, 
+        applies_once,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__discount.sql:42]: This both look like they are booleans. If so, we should rename to be consistent with our boolean naming conventions. Although, I am having a hard time turning these into `is_applies_once` 🤔. What are your thoughts?

File: models/stg_recharge__order.sql [hunk 23]
```diff
@@ -41,24 +41,24 @@ final as (
...
+        billing_address_phone, 
+        shipping_address_first_name || ' ' || shipping_address_last_name as shipping_address_full_name, 
+        shipping_address_address_1 as shipping_address_line_1, 
+        shipping_address_address_2 as shipping_address_line_2, 
+        shipping_address_city, 
+        shipping_address_province, 
+        shipping_address_country, 
+        shipping_address_zip, 
+        shipping_address_company, 
+        shipping_address_phone,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__order.sql:61]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__order_line_item.sql [hunk 24]
```diff
@@ -31,12 +31,12 @@ final as (
...
-        property_shipping_interval_unit_type as shipping_interval_unit_type,
-        property_shipping_interval_frequency as shipping_interval_frequency,
-        property_title,
-        property_price,
+        property_charge_interval_unit_type as charge_interval_unit_type, 
+        property_charge_interval_frequency as charge_interval_frequency, 
+        property_shipping_interval_unit_type as shipping_interval_unit_type, 
+        property_shipping_interval_frequency as shipping_interval_frequency, 
+        property_title, 
+        property_price,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__order_line_item.sql:39]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__payment_source.sql [hunk 25]
```diff
@@ -1,5 +1,5 @@
-{{ config(enabled=var('recharge__payment_source_enabled', True)) }}
-with base as (
+{{ config(enabled=var('recharge__payment_source_enabled', True)) }} 
+with base as (
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__payment_source.sql:2]: Just a small note that this will likely be changed to payment_method.

File: models/stg_recharge__payment_source.sql [hunk 26]
```diff
@@ -33,15 +33,15 @@ final as (
...
-        billing_address_phone,
+        billing_address_first_name || ' ' || billing_address_last_name as billing_address_full_name, 
+        billing_address_address_1 as billing_address_line_1, 
+        billing_address_address_2 as billing_address_line_2, 
+        billing_address_city as billing_address_city, 
+        billing_address_province, 
+        billing_address_zip, 
+        billing_address_country, 
+        billing_address_company, 
+        billing_address_phone,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__payment_source.sql:44]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__product.sql [hunk 27]
```diff
@@ -2,7 +2,7 @@
 with base as (
 
     select *
-    from {{ ref('stg_recharge__product_tmp') }}
+    from {{ ref('stg_recharge__product_tmp') }}
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__product.sql:5]: Just a small note that this will likely change to plan

File: models/stg_recharge__product.sql [hunk 28]
```diff
@@ -29,19 +29,19 @@ final as (
...
+        subscription_defaults_expire_after_specific_number_of_charges as expire_after_specific_number_of_charges, 
+        subscription_defaults_modifiable_properties as modifiable_properties, 
+        subscription_defaults_number_charges_until_expiration as number_charges_until_expiration, 
+        subscription_defaults_order_day_of_month as order_day_of_month, 
+        subscription_defaults_order_day_of_week as order_day_of_week, 
+        subscription_defaults_order_interval_frequency_options as order_interval_frequency_options, 
+        subscription_defaults_order_interval_unit as order_interval_unit, 
+        subscription_defaults_storefront_purchase_options as storefront_purchase_options, 
+        collection_id, 
+        shopify_product_id,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__product.sql:44]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__subscription_history.sql [hunk 29]
```diff
@@ -32,14 +32,14 @@ final as (
         price,
         quantity,
         status,
-        property_charge_interval_unit_type as charge_interval_unit_type,
+        property_charge_interval_unit_type as charge_interval_unit_type,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__subscription_history.sql:35]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/stg_recharge__subscription_history.sql [hunk 30]
```diff
@@ -32,14 +32,14 @@ final as (
...
+        property_charge_interval_unit_type as charge_interval_unit_type, 
         charge_interval_frequency,
         order_interval_unit,
         order_interval_frequency,
         order_day_of_month,
         order_day_of_week,
-        property_shipping_interval_unit_type as shipping_interval_unit_type,
-        property_shipping_interval_frequency as shipping_interval_frequency,
+        property_shipping_interval_unit_type as shipping_interval_unit_type, 
+        property_shipping_interval_frequency as shipping_interval_frequency,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_recharge__subscription_history.sql:42]: Same comment about these all being custom fields. We should let the variables handle these.

File: models/tmp/stg_recharge__address_shipping_line_tmp.sql [hunk 31]
```diff
@@ -1,5 +1,3 @@
 
-select
-    *
-from
-    {{ var('address_shipping_line') }}
\ No newline at end of file
+select *
+from {{ var('address_shipping_line') }}
```
Comment (fivetran-joemarkiewicz) [inline models/tmp/stg_recharge__address_shipping_line_tmp.sql:3]: A small style update request. Can we have the tmp models follow this style structure. Just to be consistent with our other packages.

File: packages.yml [hunk 32]
```diff
@@ -2,5 +2,5 @@
 packages:
 - package: fivetran/fivetran_utils
   version: [">=0.4.0", "<0.5.0"]
-# - package: dbt-labs/spark_utils
-#   version: [">=0.3.0", "<0.4.0"]
\ No newline at end of file
+# - package: dbt-labs/spark_utils 
+#   version: [">=0.3.0", "<0.4.0"]
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:6]: Is this intentionally left out?

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-catfritz thanks for working through this initial build! I intentionally did not make a lot of notes around the changed source tables. We can update those down as we encounter them.

I just have a few small comments, suggestions, and questions. For going through this review you can see the comments below. Feel free to apply the updates and make a note in the comment that it is applied, I can then resolve from there. If there are any items you have questions about, we can have a dialogue within the comments of this PR.

Let me know if you have any questions!

File: dbt_project.yml (hunk 10 above)
Comment (fivetran-catfritz) [reply dbt_project.yml:33]: I am having some difficulty getting this to function without the empty variables. I think this is because the variable name is passed to `fill_pass_through_columns` as a string instead of the var contents.

Do you know of any packages that I can reference? I was searching through our repo to try to find an example, and our most recent packages use the empty brackets in dbt_project.yml.

File: README.md (hunk 9 above)
Comment (fivetran-catfritz) [reply README.md:66]: Makes sense (especially now since I understand the passthroughs better)!

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/bug-report.yml:4]: done

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/config.yml:4]: done

File: .github/ISSUE_TEMPLATE/feature-request.yml (hunk 3 above)
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/feature-request.yml:4]: done

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:1]: done

File: CHANGELOG.md (hunk 5 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:4]: done

File: CHANGELOG.md (hunk 6 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:9]: done

File: integration_tests/.gitignore (hunk 11 above)
Comment (fivetran-catfritz) [reply integration_tests/.gitignore:7]: I don't have a folder called `env`, but is this meant for when others use it?

File: integration_tests/dbt_project.yml (hunk 12 above)
Comment (fivetran-catfritz) [reply integration_tests/dbt_project.yml:31]: Sorry that was an oversight, so no!

File: macros/.gitkeep (hunk 13 above)
Comment (fivetran-catfritz) [reply macros/.gitkeep:1]: Nope--deleted.

File: models/src_recharge.yml (hunk 14 above)
Comment (fivetran-catfritz) [reply models/src_recharge.yml:14]: Removed per our discussion at standup today.

File: models/stg_recharge__address.sql (hunk 15 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address.sql:36]: Thanks. Removed from models, docs, and seeds.

File: models/stg_recharge__address.sql (hunk 16 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address.sql:41]: Done.

File: models/stg_recharge__address_shipping_line.sql (hunk 17 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address_shipping_line.sql:28]: Yes, I have now added. Since this will also be added to new tables, I have added the `_fivetran_deleted ` field and filter to:

- ADDRESS
- ADRESS_SHIPPING_LINE
- CHARGES
- CUSTOMER
- ONE_TIME_PRODUCT
- ORDER
- PRODUCT
- SUBSCRIPTION
- SUBSCRIPTION_HISTORY

File: models/stg_recharge__address_shipping_line.sql (hunk 18 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address_shipping_line.sql:29]: I removed, but to make sure I understand, is my understanding correct we still want to keep `_fivetran_synced`  in the source because of [these lines](https://github.com/fivetran/dbt_recharge_source/blob/main/models/src_recharge.yml#L9) in the source yml?

File: models/stg_recharge__charge.sql (hunk 19 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__charge.sql:70]: Done.

File: models/stg_recharge__charge_line_item.sql (hunk 20 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__charge_line_item.sql:28]: `property_title` is the recharge title, and `title` is the Shopify title, so was thinking that would differentiate, but upon 2nd thought, it's really not necessary, so I reverted back to the original title to avoid confusion.

File: models/stg_recharge__charge_line_item.sql (hunk 21 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__charge_line_item.sql:39]: Removed.

File: models/stg_recharge__discount.sql (hunk 22 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__discount.sql:42]: I took a closer look, and it looks like `applies_once` is only in our sample dataset and not in the ERD, so I will just remove it. That leaves `is_once_per_customer`, which sounds ok to me.

File: models/stg_recharge__order.sql (hunk 23 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__order.sql:61]: Removed.

File: models/stg_recharge__order_line_item.sql (hunk 24 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__order_line_item.sql:39]: Removed.

File: models/stg_recharge__payment_source.sql (hunk 25 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__payment_source.sql:2]: I updated the name and can use identifiers to get around the table name issue, however I'm noticing the columns have changed. I can make mock data in our seed, but then I can't run any real data. Any suggestions how to get around/reconcile the column issue?

File: models/stg_recharge__payment_source.sql (hunk 26 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__payment_source.sql:44]: Removed.

File: models/stg_recharge__product.sql (hunk 27 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__product.sql:5]: I have updated the table names, same thing with the column names.

File: models/stg_recharge__product.sql (hunk 28 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__product.sql:44]: Removed.

File: models/stg_recharge__subscription_history.sql (hunk 29 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__subscription_history.sql:35]: Removed.

File: models/stg_recharge__subscription_history.sql (hunk 30 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__subscription_history.sql:42]: Removed.

File: models/tmp/stg_recharge__address_shipping_line_tmp.sql (hunk 31 above)
Comment (fivetran-catfritz) [reply models/tmp/stg_recharge__address_shipping_line_tmp.sql:3]: Done.

File: packages.yml (hunk 32 above)
Comment (fivetran-catfritz) [reply packages.yml:6]: Oh I had commented it out because I wasn't sure if it was necessary in the source package since it seemed to run fine without it on databricks. Should I leave it in by default?

File: dbt_project.yml (hunk 10 above)
Comment (fivetran-catfritz) [reply dbt_project.yml:33]: Oh also--is this something maybe that could be added to [fivetran_utils](https://github.com/fivetran/dbt_fivetran_utils/blob/releases/v0.4.latest/macros/fill_pass_through_columns.sql)? Having the default be that empty list instead of having no default on line 4?

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Also would we also want to reference community instead in the README too?

Comment (fivetran-joemarkiewicz) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Yup! We want to make the update here and in the references section in the README. When updating the README we should remove the office hours section with and replace it with a note about the community.

Comment (fivetran-joemarkiewicz) [reply .github/ISSUE_TEMPLATE/config.yml:4]: You can use what I added to the microsoft package. https://github.com/fivetran/dbt_microsoft_ads#-are-there-any-resources-available

File: dbt_project.yml (hunk 10 above)
Comment (fivetran-joemarkiewicz) [reply dbt_project.yml:33]: Having discussed internally, we do in fact still need to keep these default declarations in the package dbt_project.yml.

I think it would be great to open a feature request for this default variable declaration to be part of the macro.

File: integration_tests/.gitignore (hunk 11 above)
Comment (fivetran-joemarkiewicz) [reply integration_tests/.gitignore:7]: Yeah! The env is created if you ever create a virtual environment. This will store your entire environment in the folder and is not something we would want to commit to git. Adding it to the .gitignore ensures no virtual environments are committed to the repo.

File: models/stg_recharge__address_shipping_line.sql (hunk 17 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__address_shipping_line.sql:28]: Great! Thanks for adding this to the relevant models!

File: models/stg_recharge__address_shipping_line.sql (hunk 18 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__address_shipping_line.sql:29]: Great observation! The `loaded_at_field` is used to [calculate freshness](https://docs.getdbt.com/reference/resource-properties/freshness#definition). Since we have removed the freshness test, this is _technically_ not needed anymore. However, I think it would be good to still include as we may want to expand the freshness in a future update.

Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__address_shipping_line.sql:29]: This update looks good! I did have one additional question that I would be curious to understand. I know some models (plan for example) have both a `is_deleted` (I assume generated from Recharge themselves) and a `_fivetran_deleted` field (generated from Fivetran). Is there a possibility that these fields do not match?

I think going forward we should definitely filter on the _fivetran_deleted column, but curious if these two fields have potential to confuse customers if they have differing functions.

File: models/stg_recharge__charge_line_item.sql (hunk 20 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__charge_line_item.sql:28]: Thanks for the explanation! Yeah I like keeping this as `property_title` to be consistent with the source data and avoid confusion.

File: models/stg_recharge__discount.sql (hunk 22 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__discount.sql:42]: Great! Thanks for confirming and making the appropriate updates.

File: models/stg_recharge__payment_source.sql (hunk 25 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__payment_source.sql:2]: As discussed internally we will avoid the column differences for the time being. We can account for those once the ERD changes are reflected in our data set.

File: packages.yml (hunk 32 above)
Comment (fivetran-joemarkiewicz) [reply packages.yml:6]: Yeah let's leave it in as I assume the transform package will require it and we can keep this dependency in this package.

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-catfritz thanks for working through these updates! I resolved a number of the comments that look good on my end and responded to a few with some clarifications.

Let me know if you have any questions when responding to my remaining comments 😄

File: dbt_project.yml (hunk 10 above)
Comment (fivetran-catfritz) [reply dbt_project.yml:33]: Added back in, and issue submitted!

File: integration_tests/.gitignore (hunk 11 above)
Comment (fivetran-catfritz) [reply integration_tests/.gitignore:7]: Makes sense. Done.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (fivetran-catfritz) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Thank you. Done.

File: models/stg_recharge__address_shipping_line.sql (hunk 18 above)
Comment (fivetran-catfritz) [reply models/stg_recharge__address_shipping_line.sql:29]: I see what you mean. I have removed `_fivetran_deleted ` from the output of the staging models and kept it in the source for filtering.

File: packages.yml (hunk 32 above)
Comment (fivetran-catfritz) [reply packages.yml:6]: Done.

File: integration_tests/.gitignore (hunk 11 above)
Comment (fivetran-joemarkiewicz) [reply integration_tests/.gitignore:7]: @fivetran-catfritz apologies for not calling this out in the initial comment, can we also add this to the .gitignore for the integration_tests directory.

File: models/stg_recharge__address_shipping_line.sql (hunk 18 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__address_shipping_line.sql:29]: Great thanks! I do still want to know what the different between `_fivetran_deleted` and `is_deleted` are 🤔. This may be a good question for Elan.

Comment (fivetran-joemarkiewicz) [reply models/stg_recharge__address_shipping_line.sql:29]: I currently see these two fields in the [plan](https://github.com/fivetran/dbt_recharge_source/blob/main/models/stg_recharge__plan.sql) model

Comment (fivetran-catfritz) [reply models/stg_recharge__address_shipping_line.sql:29]: Looking at the [recharge docs](https://developer.rechargepayments.com/2021-01/webhooks_explained), it does appear that some of these tables use a recharge-generated deleted flag, though not all of these seem to correspond to a column in the fivetran ERD. `Plan` no long appears to have `is_deleted`, but those columns aren't finalized yet anyway. However, for `order` and `one_time_product` I think it would make sense to keep `is_deleted` in case the user wanted to filter them out?

File: integration_tests/.gitignore (hunk 11 above)
Comment (fivetran-catfritz) [reply integration_tests/.gitignore:7]: Thanks. Done.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks again for addressing these comments @fivetran-catfritz! Closing this PR.
